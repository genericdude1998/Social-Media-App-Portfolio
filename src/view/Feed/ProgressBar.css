*{
    box-sizing: border-box;
    margin: 0;
}
.progressBar {
    height: 100px;
    width: 300px;
    display: flex;
    flex-direction: column;
    justify-items: flex-end;
    align-items: center;
    font-weight: bold;
}

.bar{
    border-radius: 20px;
    height: 80%;
    width: 20%;
    background-color: white;
    animation: animate-positive 1.5s  infinite;
}

.barContainer{
    margin-top: 10px;
    border-radius: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: darkblue;
    height: 20%;
    width: 70%;
}

@keyframes animate-positive {

    33% {
      transform: translate(-50px);
    }
    66% {
        transform: translate(50px);
      }
    100% {
        transform: translate(0px);
    }
  }


.empty{

}
.mid{

}
.full{

}