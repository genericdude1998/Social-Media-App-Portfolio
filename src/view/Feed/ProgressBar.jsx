import React from 'react';

const ProgressBar = ({label, max, value}) => {
    return (
        <div>
            <label>{label}</label>
            <progress max={max} value={value}></progress>
        </div>
    );
}

export default ProgressBar;
