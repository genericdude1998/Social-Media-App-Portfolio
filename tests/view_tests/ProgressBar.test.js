import React from 'react';
import { shallow } from 'enzyme';
import ProgressBar from '../../src/view/Feed/ProgressBar';

const wrapper = shallow(
    <ProgressBar
        label={'test_label'}
    />);

it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();    
});