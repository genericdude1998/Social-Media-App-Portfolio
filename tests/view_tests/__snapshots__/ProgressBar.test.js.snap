// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="progressBar"
>
  <label>
    test_label
  </label>
  <div
    className="barContainer"
  >
    <div
      className="bar"
    />
  </div>
</div>
`;
